components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -994,13 +994,16 @@ abstract class AbstractFetchDownloadService : Service() {
             val resultContentType = if (!contentTypeFromFile.isNullOrEmpty()) {
                 contentTypeFromFile
             } else {
-                if (!contentType.isNullOrEmpty()) {
-                    contentType
-                } else {
-                    "*/*"
-                }
+                getNotEmptyMimeType(contentType)
             }
-            return (DownloadUtils.sanitizeMimeType(resultContentType) ?: "*/*")
+            return (getNotEmptyMimeType(DownloadUtils.sanitizeMimeType(resultContentType)))
+        }
+
+        @VisibleForTesting
+        internal fun getNotEmptyMimeType(mimeType: String?) = if (!mimeType.isNullOrEmpty()) {
+            mimeType
+        } else {
+            "*/*"
```

Comment:
We could re-use [getSafeContentType](https://github.com/mozilla-mobile/android-components/blob/9be7ff559413f97da02b97e0c3f315b46e647f3d/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L992) that tries to extract the `MimeType`from the physical file, what do you think?

Reply:
Adding this new safety net allows for a quick and easy way to ensure we don't get that crash anymore.
Using `getSafeContentType` would mean taking another stab at inferring the mime type but in the current implementation, as that returns `DownloadUtils.sanitizeMimeType(resultContentType) ?: "*/*"` still has the possibility of returning an empty String.
Instead of wrapping `getSafeContentType` and others in `getNotEmptyMimeType` we could add one last check for empty strings in that method and use that.

Reply:
I've updated the patch to use `getSafeContentType`.
Would appreciate if you could comment on the approach, if it's okay.
I'll work next on updating tests and documentation.